salt/states/aptpkg.py

```diff
@@ -14,11 +14,21 @@
 log = logging.getLogger(__name__)
 
 
+# Define the module's virtual name
+__virtualname__ = 'apt'
+
+
 def __virtual__():
     '''
     Only work on apt-based platforms with pkg.get_selections
     '''
-    return 'pkg.get_selections' in __salt__
+    try:
+        __salt__['pkg.get_selections']
+    except KeyError:
+        # direct access not to iterate over lazyloader
+        return False
+    else:
+        return __virtualname__
 
```

Comment:
return 'apt' if __salt__.get('pkg.get_selections', False) else False

would this not do the same thing, but not need the try except loop?

I would think the direct get should act the same, but handle it for you, so it at least looks similar to what is there currently.